### PR TITLE
upgrade for nemo@3.0.0 and selenium@3.0.0

### PR DESCRIPTION
**Pull request by kweiberth**
**What** and **Why**:

* `selenium@3.0.0` was recently released, which starts a process of deprecating selenium's built-in promise manager. More info [here](https://github.com/SeleniumHQ/selenium/issues/2969)
* Is also seems that some of the `driver` and `session` APIs have changed
	* e.g. you must now use `session.getId()` instead of `session.id_` (which was being used in this plugin)
* This PR removes dependency on selenium's promise manager
* It also fixes `getJobUrl` method, which is broken due to breaking changes in `selenium@3.0.0`
* Updated README to reflect these changes

**Other changes**:

* due to the fact that `selenium@3.0.0` requires `node@6.0.0` or later, I updated some of the syntax
	* e.g. change `var` to `let` or `const`
* Removed `Gruntfile.js` since there were no tests written at all in this repo, then added a "todo" `echo` statement for `npm test`
* Switched `_.extend` for native `Object.assign`
	* also, since we then we're using `underscore` only for `_.defaults`, pulled in `lodash.defaults` instead of full `underscore` library

**Questions**:

* should this be made into a separate `3.x` branch?

**Reviewers**:

- [x] @gkushang 
- [ ] @grawk

### CODE REVIEW

**Comment by grawk** [comment]
LGTM! @gkushang what do you think?

**Comment by gkushang** [comment]
LGTM. We will release these changes with V3 to stay consistent with Nemo

File: package.json [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
+  "version": "3.0.0",
+  "description": "updates Sauce Labs jobs with test information and test results",
   "main": "index.js",
   "author": {
     "name": "kushang gajjar",
     "email": "g.kushang@gmail.com"
   },
   "scripts": {
-    "test": "grunt"
+    "test": "echo \"TODO: write tests\""
```
Comment (gkushang) [inline package.json:11]: This module is missing Tests from beginning. We should add them (probably next PR?)

**Comment by kweiberth** [comment]
I've tested the latest `develop` which has all these changes, and everything is working well. I don't use cucumber so I can't test the cucumber tag stuff. However, we are updating the test name via `updateJob`, so that method is working. I'm also grabbing the url via `getJobUrl` and passing test result via `isJobPassed`, so it seems everything is working. See screenshots below for confirmation (url in terminal, name updated to "kurt" in saucelabs, and green checkmark in saucelabs for success result):

screen shot 2017-05-26 at 11 56 37 am

screen shot 2017-05-26 at 11 51 22 am

**Comment by kweiberth** [comment]
@gkushang my testing went well, see comment above. Let me know if you'd like me to test anything else.

**Comment by gkushang** [comment]
Thanks @kweiberth for testing the changes (you've covered all methods). Will publish it as V3 by EOD today.

**Pull request merged**